v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -682,3 +702,52 @@ npm pkg set type="module"
 
   return false;
 }
+
+/**
+ * Returns true if Hardhat is installed locally or linked from its repository,
+ * by looking for it using the node module resolution logic.
+ *
+ * If a config file is provided, we start looking for it from there. Otherwise,
+ * we use the current working directory.
+ */
+async function isHardhatInstalledLocallyOrLinked(
+  configPath: string,
+  log: debug.Debugger,
+) {
+  try {
+    // Setup a require to let us Node resolve based on the
+    // user's repo
+    const require = createRequire(configPath ?? process.cwd());
```

Comment:
This appears to be the closest ESM equivalent to:

https://github.com/NomicFoundation/hardhat/blob/0a145098cd1f50f6b46fc17ae9a19d393c56d9d3/packages/hardhat-core/src/internal/core/execution-mode.ts#L11-L27

Reply:
We have `findDependencyPackageJson` in `hardhat-utils`. Shouldn't that be enough?